--- build/rules.mk.in	2013-10-31 15:04:11.630200531 +0800
+++ build/rules.mk.in.relax	2013-10-31 15:07:18.283082057 +0800
@@ -203,20 +203,16 @@
 	$(COMPILE) -c $<
 
 .c.lo:
-	$(AI_ARCHIVE_DIR)/llvm/install/bin/clang $(ALL_CFLAGS) $(ALL_CPPFLAGS) $(ALL_INCLUDES) -emit-llvm -c $< -o $*.bc
-	$(AI_ARCHIVE_DIR)/llvm/install/bin/opt -load=$(AI_HOME)/build/llvm/AI/LLVMAIPrepare.so -AIPrepare <$*.bc >$*-Marked.bc
-	$(AI_ARCHIVE_DIR)/llvm/install/bin/opt -load=$(AI_HOME)/build/llvm/AI/LLVMAITrace.so -AITrace <$*-Marked.bc >$*-Trace.bc
-	llc $*-Trace.bc -o $*.s
+	$(AI_ARCHIVE_DIR)/llvm/install/bin/opt -load=$(AI_HOME)/build/llvm/AI/LLVMAITolerate.so -AITolerate -AISanitizedBSetFile=$(AI_ARCHIVE_DIR)/performance/apache/httpd-2.4.6/install/sanitized-bset.txt -AIRelax=1 <$*-Marked.bc >$*-Relax.bc
+	llc $*-Relax.bc -o $*.s
 	$(LIBTOOL) --mode=compile $(CC) $(PICFLAGS) $(LTCFLAGS) -c $*.s && touch $@
 
 .s.lo:
 	$(LT_COMPILE)
 
 .c.slo:
-	$(AI_ARCHIVE_DIR)/llvm/install/bin/clang $(ALL_CFLAGS) $(ALL_CPPFLAGS) $(ALL_INCLUDES) -emit-llvm -c $< -o $<.bc -o $*.bc
-	$(AI_ARCHIVE_DIR)/llvm/install/bin/opt -load=$(AI_HOME)/build/llvm/AI/LLVMAIPrepare.so -AIPrepare <$*.bc >$*-Marked.bc
-	$(AI_ARCHIVE_DIR)/llvm/install/bin/opt -load=$(AI_HOME)/build/llvm/AI/LLVMAITrace.so -AITrace <$*-Marked.bc >$*-Trace.bc
-	llc $*-Trace.bc -o $*.s
+	$(AI_ARCHIVE_DIR)/llvm/install/bin/opt -load=$(AI_HOME)/build/llvm/AI/LLVMAITolerate.so -AITolerate -AISanitizedBSetFile=$(AI_ARCHIVE_DIR)/performance/apache/httpd-2.4.6/install/sanitized-bset.txt -AIRelax=1 <$*-Marked.bc >$*-Relax.bc
+	llc $*-Relax.bc -o $*.s
 	$(LIBTOOL) --mode=compile $(CC) -DLINUX -D_REENTRANT -D_GNU_SOURCE  $(ALL_INCLUDES) $(SHLTCFLAGS) -c $< && touch $@
 
 .$(CXX_SUFFIX).lo:
